src/index.ts

```diff
@@ -197,23 +197,37 @@ export class IDX {
     }
   }
 
-  async _getIDXDoc(did: string): Promise<Doctype> {
-    const doc = await this._ceramic.createDocument(
+  async _createIDXDoc(did: string): Promise<Doctype> {
+    return await this._ceramic.createDocument(
       'tile',
       {
         deterministic: true,
-        metadata: { controllers: [did], family: 'IDX', schema: schemas.IdentityIndex },
+        metadata: { controllers: [did], family: 'IDX' },
       },
       { anchor: false, publish: false }
     )
+  }
+
+  async _getIDXDoc(did: string): Promise<Doctype> {
+    const doc = await this._createIDXDoc(did)
     if (doc.metadata.schema !== schemas.IdentityIndex) {
       throw new Error('Invalid document: schema is not IdentityIndex')
     }
     return doc
   }
 
   async _getOwnIDXDoc(): Promise<Doctype> {
-    return await this._getIDXDoc(this.id)
+    const doc = await this._createIDXDoc(this.id)
+    if (doc.metadata.schema == null) {
+      // Doc just got created, need to update it with schema
+      await doc.change(
+        { metadata: { ...doc.metadata, schema: schemas.IdentityIndex } },
+        { anchor: true, publish: true }
+      )
+    } else if (doc.metadata.schema !== schemas.IdentityIndex) {
+      throw new Error('Invalid document: schema is not IdentityIndex')
+    }
```

Comment:
Should we not just change to the correct schema in this case?

Reply:
I don't think we can because we have no idea of the changes in the schema itself?
I think if we make a breaking schema change, we'd have more logic here so we can upgrade to the new version of the schema knowing how it affects the contents, but that's not something we can anticipate here.